### DISCUSSION TITLE
Username with Domain

### DISCUSSION THREAD

**Discussion by IdleThought** [ing-bank/cassandra-jdbc-wrapper#73] [Q&A]
Hello:

I'm using Liquibase 4.25 so I wanted to use the 4.25 Cassandra wrapper, but ran into issue with a username with a backslash after the domain.

If I use my username that doesn't need a backslash, everything works fine, but when I use a username with domain\xxxx then the backslash becomes an invalid character for the driver. I have tried making the backslash the Unicode %5C but that also didn't work.

In IntelliJ if I create a Cassandra connection with this driver and tell it to use a username and password, then supply the username (that includes the domain and backslash) and password, and then omit the username and password from the JDBC URL, it works fine that way too.

Not sure what to do here, I sadly _**have**_ to supply a JDBC URL with username (including domain) and password in it for my CI/CD pipeline at my company, but I cannot figure out how to get around this backslash issue.

Thank you for any assistance you might be able to provide.

**Comment by maximevw** [comment]
Hello @IdleThought,

In all cases, you have to use the URL-encoded backslash `%5C` in the JDBC URL since this one is checked to be a valid URI, so characters as backslashes are not allowed without URL-encoding.

Could you be more specific about the error you got when the connection failed with the username including domain, especially with the URL-encoded backslash?

On my side, I tried the following configuration:
Liquibase 4.29.2 (for your case, it should not change much to use prior version, but I recommend to upgrade if possible).
Cassandra JDBC Wrapper 4.12.0

Using Cassandra 5 image:
`docker run -p 9042:9042 --name cassandra5 -d cassandra:5`

In cassandra.yaml:
```yaml
authenticator:
  class_name : org.apache.cassandra.auth.PasswordAuthenticator
authorizer: CassandraAuthorizer
role_manager: CassandraRoleManager
```

In my database, I created the following role with the appropriate permissions:
`CREATE ROLE 'domain\myuser' WITH PASSWORD = '{password}' WITH LOGIN;`

In liquibase.properties:
```properties
liquibase.command.url: jdbc:cassandra://localhost:9042/testkeyspace?localdatacenter=datacenter1&user=domain%5cmyuser&password={password}
```

And with this configuration, I successfully ran these commands:
```
liquibase status
liquibase update-sql
liquibase update
```

**Reply by IdleThought** [reply]
Thanks you for your response, I was able to correct my issue by not specifying the password as a part of the jdbc url itself. My password also had a special character in it, which was throwing an invalid character error. But once I started using --password=domain\xxx --username=xxx everything is working fine for me.
